addons/point_of_sale/static/src/js/screens.js

```diff
@@ -2192,8 +2238,18 @@ var PaymentScreenWidget = ScreenWidget.extend({
                 self.gui.show_screen('receipt');
             });
         } else {
-            this.pos.push_order(order);
-            this.gui.show_screen('receipt');
+            var ordered = this.pos.push_order(order);
+            ordered.then(function() {
+                self.gui.show_screen('receipt');
+            });
+            ordered.catch(function(value) {
+                order.set_to_email(false);
+                self.gui.show_screen('receipt');
+                self.gui.show_popup('error',{
+                    'title': "Error: no internet connection",
+                    'body':  "There is no internet connection, impossible to send the email. The ticket will be print.",
```

Comment:
'will be printed'. Not sure about the message, are you sure the ticket will be printed, or the cashier will have to 'unthick' the checkbox send by mail to print it.